gradle-sls-packaging/src/test/groovy/com/palantir/gradle/dist/service/JavaServiceDistributionPluginTests.groovy

```diff
@@ -354,10 +354,38 @@ class JavaServiceDistributionPluginTests extends GradleIntegrationSpec {
         manifest.productName() == "service-name"
 
         // check files in deployment/ copied successfully
-        String actualConfiguration = file('dist/service-name-0.0.1/deployment/configuration.yml').text
+        String actualConfiguration = new File(projectDir, 'dist/service-name-0.0.1/deployment/configuration.yml').text
         actualConfiguration == deploymentConfiguration
     }
 
+    def 'allows another task to produce configuration.yml'() {
+        given:
+        createUntarBuildFile(buildFile)
+        debug = true
+
+        // language=Gradle
+        buildFile << '''
+            task createConfigurationYml {
+                outputs.file('build/some-place/configuration-but-called-something-else.yml')
+                
+                doFirst {
+                    file('build/some-place/configuration-but-called-something-else.yml').text = 'custom: yml'
+                }
+            }
+
+            distribution {
+                configurationYml.fileProvider(tasks.named('createConfigurationYml').map { it.outputs.files.singleFile }) 
```

Comment:
In java plugin code you're just going be doing `taskProvider.flatMap(CustomTask::getOutputFile)`, so not this ugly.